File: codility/Fish.js
function solution(A, B) {
    // Implement your solution here[0,1,1,0,0]
    let alive = 0;
    const downFish = [];
    for (let i = 0; i < B.length; i++) {
        const v = B[i];
        if (v === 0) {
            if (downFish.length === 0) {
                alive++;
            } else {
                while (downFish[downFish.length - 1] < A[i]) {
                    downFish.pop();
                    if (downFish.length === 0) {
                        alive++;
                    }
                }
            }
        } else {
            downFish.push(A[i]);
        }
    }
    // console.log(downFish, alive)
    return downFish.length + alive;
}

File: codility/python/Fish.py
def solution(A, B):
    # Implement your solution here
    alive = 0
    down_fish = []
    for i in range(len(B)):
        b = B[i]
        if b == 0:
            if len(down_fish) == 0: 
                alive += 1
            else:
                while down_fish[-1] < A[i]:
                    down_fish.pop()
                    if len(down_fish) == 0:
                        alive += 1
                        break
        else:
            down_fish.append(A[i])
    
    return len(down_fish) + alive